lib/puffy/resolver.rb

```diff
@@ -56,6 +56,14 @@ def resolv_apt_mirror(url)
       res
     end
 
+    def resolv_azure_ip_range(service_name)
+      # https://www.microsoft.com/en-us/download/details.aspx?id=56519
+      @azure_ip_range ||= JSON.parse(URI('https://download.microsoft.com/download/7/1/D/71D86715-5596-4529-9B13-DA13A5DE5B63/ServiceTags_Public_20240701.json').read)
```

Comment:
This URL contains a date, we should use a generic one, if available.

Reply:
Crap!

They have other ways of getting this list but it require a subscription ID :fearful: :
https://learn.microsoft.com/en-us/azure/virtual-network/service-tags-overview#use-the-service-tag-discovery-api

Reply:
Let's see what the URI look like in one week.  If the UID is stable and only the date at the end change every 7 days, we can compute it.

In fact we only have to wait 2 days :-).